base/strings/string.jl

```diff
@@ -491,7 +493,8 @@ end
     @inbounds length_continued(s, i, j, c)
 end
 
-@assume_effects :terminates_locally @inline @propagate_inbounds function length_continued(s::String, i::Int, n::Int, c::Int)
+# duck-type s so that external UTF-8 string packages like StringViews can hook in
+@assume_effects :terminates_locally @inline @propagate_inbounds function length_continued(s, i::Int, n::Int, c::Int)
```

Comment:
This `@assume_effects` is invalid without the type assert.

Reply:
Okay, reverted that function.